### ISSUE TITLE
"Broadcast from ..." Has appeared

### ISSUE THREAD

**Issue by DanDon** [greghesp/assistant-relay#67]
I'm guessing a Google server side change as it just started happening and possibly something you are unable to do anything about but it ruins the setup for leaving doors open etc if there if an annoying "broadcast from USER" before everything :(

**Comment by Webunity** [comment]
Yep, same thing happened here as well. Widely irritating. If i could just make it say "Broadcast from HOUSE" etc...

**Comment by DanDon** [comment]
You can.

Add a new Google account that you create to all of your Google home speakers and just make the first name of that account be "home"

**Comment by coreypmurphy** [comment]
Also experiencing now in the last day or so. Very annoying! Guess that's what happens when we depend on an API we don't have control overb and they change something.

**Comment by mhofman** [comment]
The "broadcast from xxx" only seem to be added when using a text command and not a voice command. Has anyone tried to do a TTS on the text and send that along with the text command?
We'd loose the nice Google assistant voice though.

**Comment by DusBus** [comment]
Sadness, I guess Google is fiddling with the API again. I remember a guy called "josharmour" from another bug thread (he was one of the google dev's or something), maybe he'd be able to help or provide a workaround of some sort?

**Comment by DanDon** [comment]
I'm just using cast-web now with smart things, as the webcore smartapp allows direct casting of messages without the "broadcast from..." 
... and it even allows you to individually select specific speakers 😁

**Comment by alxbouchard** [comment]
> I'm just using cast-web now with smart things, as the webcore smartapp allows direct casting of messages without the "broadcast from..."
> ... and it even allows you to individually select specific speakers 😁

Can you elaborate? What command do you post?
Thanks!

**Comment by ryancasler** [comment]
Yes...but unfortunately that also has a very bad side effect.  If you are already casting any content to the device, after your announcement, your media doesn't restart.  With the broadcast protocol, it continues to play whatever media was being played before the announcement.  That's the benefit of using the Broadcast protocol.  I am using Cast-Web for other things, and it is great.  But I'd like the flexibility to use both.  And those silly "Broadcast from" messages are really annoying.

For those interested, cast web is maintained here:
https://github.com/vervallsweg/cast-web-api

**Comment by DanDon** [comment]
This is technically not true as the webCore integration allows you to select the ability to speak text and then resume anything that may have currently been playing.
This means I get the best of all worlds. 
The webCore sends to cast-web which then cast-web sends to Google home speakers. You can select individual speakers and the text and whether to speak and then resume.

**Comment by mhofman** [comment]
For a lot of people any interruption of the current cast is not an option. There are simply too many edge cases that aren't handled right with a solution like cast-web.

However I'm speculating the TTS portion of cast web could be used to generate the voice part of the command to send to the assistant broadcast API.

**Comment by ryancasler** [comment]
No, actually webCoRE does not allow you to resume anything that was playing previously.  That functionality is there in webCoRE for the SmartThings Sonos integration.  There is no way for the google home speakers to resume playing something if something else is "Cast" to it.  That's why the broadcast feature is so great.  It doesn't interrupt what is already playing.  But if you are casting media to your speaker from your phone or just on the speaker itself through voice commands and you issue a TTS announcement to the speaker using the CAST function (not broadcast) you CANNOT result what was previously playing.  It just doesn't work that way.  And if you are able to get it to do that under those conditions then you are a miracle worker.
